core/templates/dev/head/components/share/SharingLinksDirective.js

```diff
@@ -22,39 +22,51 @@ oppia.directive('sharingLinks', [function() {
     scope: {
       layoutType: '@',
       layoutAlignType: '@',
+      shareType: '@',
       getTwitterText: '&twitterText',
-      getExplorationId: '&explorationId'
+      getExplorationId: '&explorationId',
+      getCollectionId: '&collectionId'
     },
     templateUrl: 'components/sharingLinks',
     controller: [
-      '$scope', '$window', 'oppiaHtmlEscaper', 'ExplorationEmbedButtonService',
+      '$scope', '$window', '$injector', 'oppiaHtmlEscaper',
       'siteAnalyticsService', 'UrlInterpolationService',
       function(
-          $scope, $window, oppiaHtmlEscaper, ExplorationEmbedButtonService,
+          $scope, $window, $injector, oppiaHtmlEscaper,
           siteAnalyticsService, UrlInterpolationService) {
-        $scope.explorationId = $scope.getExplorationId();
+        if ($scope.shareType === 'exploration') {
+          $scope.explorationId = $scope.getExplorationId();
+          var ExplorationEmbedButtonService = $injector.get(
+            'ExplorationEmbedButtonService');
 
-        $scope.registerShareExplorationEvent = function(network) {
-          siteAnalyticsService.registerShareExplorationEvent(network);
-        };
+          $scope.registerShareExplorationEvent = function(network) {
+            siteAnalyticsService.registerShareExplorationEvent(network);
+          };
 
-        $scope.showEmbedExplorationModal = (
-          ExplorationEmbedButtonService.showModal);
+          $scope.showEmbedExplorationModal = (
+            ExplorationEmbedButtonService.showModal);
+        } else {
```

Comment:
else if $scope.shareType === 'collection'

... and then later ...

else throw Error(...)